electron_app/DownloadManager.js

```diff
@@ -1,29 +1,48 @@
 const Downloader = require('./Downloader');
 const path = require('path');
-const { handleFolder } = require('./helpers/handleDisk');
+const { handleFolder, fileAlreadyExists } = require('./helpers/handleDisk');
 const downloadStats = require('./helpers/downloadInfo').downloadStats;
 const handleEvents = require('./helpers/handleEvents');
 
 let filePath = '';
 let allDownloads = [];
+let fileExists = true;
 
-const downloadManager = (items, access_token, win) => {
+const downloadManager = (items, access_token, electronWin) => {
   items.forEach(item => {
-    if (item.preserveStructure) {
-      const structurePath = item.path.substring(item.path.lastIndexOf('/'), 0);
-      handleFolder(path.join(item.destination, structurePath)).then(result => {
-        allDownloads.push(processDownload(access_token, item, result, win));
-      });
-    } else {
-      allDownloads.push(processDownload(access_token, item, item.destination, win));
-    }
+    let count = 0;
+    let fileName = item.name;
+    const destination = item.preserveStructure ?
+      path.join(item.destination, item.path.substring(item.path.lastIndexOf('/'), 0)) : item.destination;
+    do {
+      fileExists = fileAlreadyExists(path.join(destination, fileName));
+      if (fileExists) {
+        count++;
+        fileName = item.name.substring(0, item.name.indexOf('.')) + '(' + (count) + ')' +
+          item.name.substring(item.name.indexOf('.'));
+      } else {
+        item.name = fileName;
```

Comment:
One last thing here, I think it would be clearer to move this `else` clause outside the `do/while` loop since it only executes once, when the loop is complete. Having it inside the `do/while` makes it seem like it might be executed more than once.